src/app/pages/system/advanced/advanced.component.ts

```diff
@@ -246,42 +252,61 @@ export class AdvancedComponent implements OnDestroy {
 
   afterInit(entityEdit: any) {
     this.entityForm = entityEdit;
-    this.swapondrive = _.find(this.fieldConfig, { 'name': 'swapondrive' });
-    this.swapondrive_subscription = entityEdit.formGroup.controls['swapondrive'].valueChanges.subscribe((value) => {
-      if (parseInt(value) === 0) {
-        this.swapondrive.warnings = helptext_system_advanced.swapondrive_warning;
-      } else {
-        this.swapondrive.warnings = null;
-      }
-    });
-
-    this.ws.call(this.queryCall).subscribe((adv_values)=>{
-      entityEdit.formGroup.controls['sed_passwd2'].setValue(adv_values.sed_passwd);
-    })
-    this.adv_serialport =
-    _.find(this.fieldConfig, { 'name': 'serialport' });
-    this.adv_serialspeed =
-    _.find(this.fieldConfig, { 'name': 'serialspeed' });
-    this.adv_serialconsole =
-    entityEdit.formGroup.controls['serialconsole'];
-    this.adv_serialspeed['isHidden'] = !this.adv_serialconsole.value;
-    this.adv_serialport['isHidden'] = !this.adv_serialconsole.value;
-    this.adv_serialconsole_subscription = this.adv_serialconsole.valueChanges.subscribe((value) => {
-      this.adv_serialspeed['isHidden'] = !value;
-      this.adv_serialport['isHidden'] = !value;
-    });
-    entityEdit.ws.call('system.advanced.serial_port_choices').subscribe((serial_port_choices)=>{
-      for(const k in serial_port_choices){
-        this.adv_serialport.options.push(
-          {
-            label: k, value: serial_port_choices[k]
+    this.ws.call('system.is_freenas').subscribe((res)=>{
+      this.is_freenas = res;
+      _.find(this.fieldConfig, { 'name': 'legacy_ui' })['isHidden'] = this.is_freenas;
+      this.swapondrive = _.find(this.fieldConfig, { 'name': 'swapondrive' });
+      this.swapondrive_subscription = entityEdit.formGroup.controls['swapondrive'].valueChanges.subscribe((value) => {
+        if (parseInt(value) === 0) {
+          this.swapondrive.warnings = helptext_system_advanced.swapondrive_warning;
+        } else {
+          this.swapondrive.warnings = null;
+        }
+      });
+      setTimeout(() => {
+        entityEdit.formGroup.controls['legacy_ui'].valueChanges.subscribe((value) => {
+          if (value) {
+            this.dialog.confirm('Warning', 'The Legacy UI is deprecated and may \
```

Comment:
This should use the helptext string rather than repeat a literal string. And the button is now "I accept the risks", different from the one above.

Reply:
Regarding the button message, I'm changing the two dialogs that lead to the legacy UI to 'Continue to the..." but this one is different because it is a checkbox chainging a setting, not actually leading to the UI